app/design/frontend/base/default/template/oyst/oneclick.phtml

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * This file is part of Oyst_OneClick for Magento.
+ *
+ * @license http://www.apache.org/licenses/LICENSE-2.0 Apache License, Version 2.0
+ * @author Oyst <plugin@oyst.com> <@oyst>
+ * @category Oyst
+ * @package Oyst_OneClick
+ * @copyright Copyright (c) 2017 Oyst (http://www.oyst.com)
+ */
+
+/**
+ * @var Oyst_OneClick_Block_OneClick $this
+ */
+?>
+<?php
+/** @var Mage_Catalog_Model_Product $product */
+$product = $this->getProduct();
+
+/** @var string $oneClickUrl */
+$oneClickUrl = $this->getOneClickUrl();
+
+/** @var Oyst_OneClick_Helper_Catalog_Data $catalogHelper */
+$catalogHelper = Mage::helper('oyst_oneclick/catalog_data');
+
+if ($catalogHelper->isSupportedProduct($product)) :
+?>
+    <div id="oyst-1click-button"></div>
+    <script>
+        (function () {
+            window.__OYST__ = window.__OYST__ || {};
+            window.__OYST__.getOneClickURL = function (cb, opts) {
+                opts = opts || {};
+
+                ready(function () {
```

Comment:
what is `ready` function? I mean, where is it declared?

Reply:
the same, where does it come from ?

Reply:
from `skin/frontend/base/default/js/oyst`: [here](https://github.com/oystparis/oyst-1click-magento/blob/v0.1.0/skin/frontend/base/default/js/oyst/oneclick.js)